app/models/course.rb

```diff
@@ -1,63 +1,29 @@
 class Course < ApplicationRecord
```

Comment:
We have now entirely removed references to EECF UUIDs from the model, these were previously in the code to act as a translation layer to obtain identifiers which were used to link to LeadProviders. Now that identifiers are right on the model, that can go.

Along with that COURSE_NAMES can go. That was being used as a pseudo-identifier as a way to link course records into the codebase. If you look you'll see lines like `ASO: "Additional Support Offer for new headteachers`. This allowed us to do things like `Course.find_by(code: :ASO)` to pull specific records out. Now that identifiers are properly in place in the DB we don't need to use this code->name translation layer and can instead query directly on identifiers. Like we are now doing in the scopes below.

Reply:
COURSE_ECF_ID_TO_IDENTIFIER_MAPPING was used in the LeadProviders model but now instead we are using the identifiers as a key for filtering available providers.

It would still be ideal to remove that and replace it with a proper relationship between the two tables.